### PR TITLE
Refactor Transaction and add settlement function

### PR DESCRIPTION
**Pull request by djleong01**
- Refactor code in Transaction.java to use ArrayList instead of Hashmap
- Rename attributes in subtransaction to improve overall code clarity
- Change input format for adding transaction
- Add settlement function to settle/repay debt for a person

### CODE REVIEW

File: src/main/java/longah/LongAh.java [hunk 1]
```diff
@@ -35,6 +35,33 @@ public void listAllDebts() {
         }
     }
 
+    /**
+     * Settles up the debts of the specified borrower by creating a transaction
+     * to pay a single person in the group.
+     * @param borrowerName The name of the borrower to settle up.
+     */
+    public void settleUp(String borrowerName) {
```
Comment (1simjustin) [inline src/main/java/longah/LongAh.java:43]: i have no idea what this is doing and putting it in main file does not help, could u chuck it into where it belongs, and isn't jx doing settlement

File: src/main/java/longah/LongAh.java [hunk 2]
```diff
@@ -57,8 +84,13 @@ public static void main(String[] args) {
                     transactions.add(parts[1], members);
                     break;
                 case "listdebts":
-                    app.listAllDebts();
-                    break;
+                    if (transactions.getTransactionListSize() == 0) {
```
Comment (1simjustin) [inline src/main/java/longah/LongAh.java:87]: do this handling using exceptions in listalldebts

File: src/main/java/longah/LongAh.java [hunk 3]
```diff
@@ -57,8 +84,13 @@ public static void main(String[] args) {
...
+                        System.out.println("No transactions found.");
+                        break;
+                    } else {
+                        app.listAllDebts();
+                        break;
+                    }
                 case "listtransactions":
                     System.out.println(transactions.listTransactions());
```
Comment (1simjustin) [inline src/main/java/longah/LongAh.java:95]: no recollection of how this was written, similarly might need a len 0 check

File: src/main/java/longah/LongAh.java [hunk 4]
```diff
@@ -100,12 +132,21 @@ public static void main(String[] args) {
                 case "listmembers":
                     members.listMembers();
                     break;
+                case "settleup":
+                    if (parts.length == 2) {
+                        String name = parts[1];
+                        app.settleUp(name);
+                    } else {
+                        System.out.println("Invalid command format. Use 'settleup NAME'");
```
Comment (1simjustin) [inline src/main/java/longah/LongAh.java:140]: exceptions exceptions

File: src/main/java/longah/LongAh.java [hunk 5]
```diff
@@ -100,12 +132,21 @@ public static void main(String[] args) {
...
+                    }
+                    break;
                 case "exit":
                     System.exit(0);
                     return;
                 default:
-                    System.out.println("Invalid command. Use 'add', 'list', 'delete', 'find', 'clear'," +
-                            " 'addmember', 'listmembers', or 'exit'.");
+                    System.out.println("Invalid command. Use 'add', 'listdebts', 'listtransactions'," +
```
Comment (1simjustin) [inline src/main/java/longah/LongAh.java:147]: exceptions exceptions (maybe)

File: src/main/java/longah/node/Transaction.java [hunk 6]
```diff
@@ -21,95 +22,100 @@ public class Transaction {
      * @throws LongAhException If the user input is in an invalid format or value.
      */
     public Transaction(String userInput, MemberList memberList) throws LongAhException {
-        // User input format: p/[person owed] p/[person1 owing] a/[amount1] p/[person2 owing] a/[amount2] ...
+        // User input format: [Lender] p/[Borrower1] a/[amount1] p/[Borrower2] a/[amount2] ...
         String[] splitInput = userInput.split("p/");
-
-        if (splitInput.length < 3 || !splitInput[0].isEmpty()) {
+        if (splitInput.length < 2 || splitInput[0].isEmpty()) {
```
Comment (1simjustin) [inline src/main/java/longah/node/Transaction.java:27]: no more or condition, name check will be done when calling memberlist

File: src/main/java/longah/node/Transaction.java [hunk 7]
```diff
@@ -21,95 +22,100 @@ public class Transaction {
...
      * @param memberList The list of members in the group.
-     * @return The amount owed by the payee.
+     * @return The amount owed by the borrower.
      * @throws LongAhException If the expression is in an invalid format or value.
      */
-    public Double addPayee(String expression, MemberList memberList) throws LongAhException {
-        String[] splitPersonOwing = expression.split("a/");
-        if (splitPersonOwing.length != 2) {
+    public Double addBorrower(String expression, MemberList memberList) throws LongAhException {
+        String[] splitBorrower = expression.split("a/");
```
Comment (1simjustin) [inline src/main/java/longah/node/Transaction.java:54]: `borrowNameAmount` would be clearer

File: src/main/java/longah/util/MemberList.java [hunk 8]
```diff
@@ -143,27 +143,27 @@ public ArrayList<Subtransaction> solveTransactions() {
 
             // Check the current pair for which balance is greater or if equal
             if (positiveBalance > negativeBalance) {
-                Subtransaction subtransaction = 
-                        new Subtransaction(negativeMember, positiveMember,
-                        negativeBalance);
+                Subtransaction subtransaction =
+                        new Subtransaction(positiveMember, negativeMember,
+                                negativeBalance);
```
Comment (1simjustin) [inline src/main/java/longah/util/MemberList.java:148]: i not sure about coding standard here, shouldn't it follow the already indented indentation

**Review by 1simjustin — COMMENTED** [review]
while ur doing this help me add an exception for member creation if name==""
some formats I'm not too sure about and a few things I cant put my finger on that feels off but ill fix it later on

**Comment by djleong01** [comment]
Refactor the user input portion to handle exceptions in the class functions rather than the main loop, @FeathersRe not sure if it affected your JUnit portion of code so you might want to take a look at it. 

Moved settlement code to transactionList rather than main class & made changes as per review above.

**Review by 1simjustin — APPROVED** [review]
LGTM, will require a little refactoring here and there but nothing urgent

**Pull request merged**